### DISCUSSION TITLE
Extensions work arounds?

### DISCUSSION THREAD

**Discussion by djpiper28** [protobuf-c/protobuf-c#592] [Q&A]
From this https://github.com/protobuf-c/protobuf-c/wiki/Extensions I understand that extensions are not supported, are there any nice work arounds for this such as splitting the blob and calling the apt unpack methods and vice versa?

**Comment by lipnitsk** [comment]
Extensions are deprecated in proto3. See
https://developers.google.com/protocol-buffers/docs/proto3#any for
alternatives. There are no plans to support proto2 extensions.

**Comment by djpiper28** [comment]
> Extensions are deprecated in proto3. See https://developers.google.com/protocol-buffers/docs/proto3#any for alternatives. There are no plans to support proto2 extensions.
> […](#)
> On Fri, Feb 5, 2021, 06:54 Danny Piper ***@***.***> wrote: From this https://github.com/protobuf-c/protobuf-c/wiki/Extensions I understand that extensions are not supported, are there any nice work arounds for this such as splitting the blob and calling the apt unpack methods and vice versa? — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub <#454>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AADLEAN3I3W6BZFJM3JIDL3S5QBBNANCNFSM4XE4VRVQ> .

without a rewrite of the source to proto3 is there any sort of janky hacky way to get extensions working?

**Comment by lipnitsk** [comment]
There are no plans to support proto2 extensions.

**Comment by xloem** [comment]
Hey @lipnitsk, I notice your reply here did not address the question asked.  It would be helpful to have information in the README and documentation regarding the policy to not support proto2 extensions.

Honestly it would be quite helpful to have information on why they were never supported, and how work was left when that decision was made, for people who run into the issue to make their own decisions moving forward.

There are still a lot of legacy proto2 files lying around.
